django/thunderstore/api/cyberstorm/views/package_preview.py

```diff
@@ -0,0 +1,288 @@
+        cache_vary += f".{sorted(params['excluded_categories'])}"
+        cache_vary += f".{params.get('section', '-')}"
+        cache_vary += f".{params.get('q', '-')}"
+        cache_vary += f".{params['ordering']}"
+        cache_vary += f".{params['page']}"
+        cache_vary += f".{params['page_size']}"
+        return cache_vary
+
+    def get_queryset(self, params) -> QuerySet[PackageListing]:
+        # Prep Q objects
+        filter_q = Q()
+        exclude_q = Q()
+
+        # Add majority of filters
+        if params.get("community_identifier", None):
+            filter_q.add(Q(community__identifier=params["community_identifier"]), Q.AND)
+
+        if params.get("namespace_identifier", None):
+            filter_q.add(
+                Q(package__namespace__name=params["namespace_identifier"]), Q.AND
+            )
+
+        if params.get("team_identifier", None):
+            filter_q.add(Q(package__owner__name=params["team_identifier"]), Q.AND)
+
+        if params.get("user_identifier", None):
+            filter_q.add(
+                Q(
+                    package__owner__members__user__username__icontains=params[
+                        "user_identifier"
+                    ]
+                ),
+                Q.AND,
+            )
+
+        if params.get("package_identifier", None):
+            filter_q.add(Q(package__name=params["package_identifier"]), Q.AND)
+
+        if not params.get("deprecated", None):
+            exclude_q.add(Q(package__is_deprecated=True), Q.OR)
+
+        if not params.get("nsfw", None):
+            exclude_q.add(Q(has_nsfw_content=True), Q.OR)
+
+        if params.get("q", None):
+            icontains_query = Q()
+            parts = [x for x in params["q"].split(" ") if x]
+
+            for part in parts:
+                for field in (
+                    "package__name",
+                    "package__owner__name",
+                    "package__latest__description",
+                ):
+                    icontains_query &= ~Q(**{f"{field}__icontains": part})
+            exclude_q.add(icontains_query, Q.OR)
+
+        # Add category filters/excludes
+        included = []
+        excluded = []
+        if params.get("section", None):
+            try:
+                section = PackageListingSection.objects.prefetch_related(
+                    "require_categories", "exclude_categories"
+                ).get(slug=params["section"])
+            except PackageListingSection.DoesNotExist:
+                pass
+            else:
+                [included.append(s.slug) for s in section.require_categories.all()]
+                [excluded.append(s.slug) for s in section.exclude_categories.all()]
+
+        if params.get("included_categories", None):
+            included = list(chain(included, params["included_categories"]))
+
+        if params.get("excluded_categories", None):
+            excluded = list(chain(excluded, params["excluded_categories"]))
```

Comment:
There's no conditional needed, just default to an empty list with `params.get("included_categories", [])`.

Also turn them into a set instead of a chained list, since there might be duplicates.